clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -673,28 +673,30 @@ private KafkaConsumer(ConsumerConfig config, Deserializer<K> keyDeserializer, De
             GroupRebalanceConfig groupRebalanceConfig = new GroupRebalanceConfig(config,
                     GroupRebalanceConfig.ProtocolType.CONSUMER);
 
-            this.groupId = groupRebalanceConfig.groupId;
+            this.groupId = Optional.ofNullable(groupRebalanceConfig.groupId);
             this.clientId = buildClientId(config.getString(CommonClientConfigs.CLIENT_ID_CONFIG), groupRebalanceConfig);
 
             LogContext logContext;
 
             // If group.instance.id is set, we will append it to the log context.
             if (groupRebalanceConfig.groupInstanceId.isPresent()) {
                 logContext = new LogContext("[Consumer instanceId=" + groupRebalanceConfig.groupInstanceId.get() +
-                        ", clientId=" + clientId + ", groupId=" + groupId + "] ");
+                        ", clientId=" + clientId + ", groupId=" + groupId.orElse("null") + "] ");
```

Comment:
Perhaps we can improve this. If the groupId is null, perhaps we can leave it out of the log context?

Reply:
I admit that 'null' is not flattering, but I was preserving the current behavior.  Would you like something else?  We can improve this is a follow-up JIRA if you'd like to discuss.

Reply:
We can do it separately if you like.